pkg/health/server/prober.go

```diff
@@ -345,9 +356,9 @@ func newProber(s *Server, nodes nodeMap) *prober {
 
 	prober.setNodes(nodes)
 	prober.OnRecv = func(addr *net.IPAddr, rtt time.Duration) {
-		prober.RLock()
```

Comment:
Consider changing the RWMutex to Mutex if there are not as many reads as before.

Reply:
The original idea behind the RWMutex rather than Mutex was that the health server itself performs writes whenever new probe results come in, while any API requests don't need write privileges to serve a response. Though we don't expect many clients to make requests, I don't see harm in leaving this the way it is.